Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp

```diff
@@ -2011,14 +2011,19 @@ void WebChromeClient::hasStorageAccess(RegistrableDomain&& subFrameDomain, Regis
         completionHandler(false);
 }
 
-void WebChromeClient::requestStorageAccess(RegistrableDomain&& subFrameDomain, RegistrableDomain&& topFrameDomain, LocalFrame& frame, StorageAccessScope scope, CompletionHandler<void(RequestStorageAccessResult)>&& completionHandler)
+void WebChromeClient::requestStorageAccess(RegistrableDomain&& subFrameDomain, RegistrableDomain&& topFrameDomain, LocalFrame& frame, StorageAccessScope scope, bool hasOrShouldIgnoreUserGesture, CompletionHandler<void(RequestStorageAccessResult)>&& completionHandler)
 {
     auto webFrame = WebFrame::fromCoreFrame(frame);
     ASSERT(webFrame);
-    if (RefPtr page = m_page.get())
-        page->requestStorageAccess(WTFMove(subFrameDomain), WTFMove(topFrameDomain), *webFrame, scope, WTFMove(completionHandler));
-    else
-        completionHandler({ });
+
+    RefPtr page = m_page.get();
+    if (!page)
+        return completionHandler({ StorageAccessWasGranted::No, StorageAccessPromptWasShown::No, scope, WTFMove(topFrameDomain), WTFMove(subFrameDomain) });
+
+    if (page->usesEphemeralSession() && !hasOrShouldIgnoreUserGesture)
```

Comment:
We support requesting storage access within an ephemeral session. Is this need for another reason?

Reply:
I needed this when we were checking state on disk, which wouldn't exist for an ephemeral session. I updated the patch to check in-memory state instead, so we can remove it now.